From f7672e3bf0a8b233df9e41c53140fd459f0d411f Mon Sep 17 00:00:00 2001
From: "Yang, Dong" <dong.yang@intel.com>
Date: Fri, 27 Sep 2019 09:23:38 +0800
Subject: [PATCH 1180/1214] enable VTD WA camera, disable for FBC and Scanout

On BXT board, the VTD WA will be applied in EVMM
so the FBC and Scanout WA is not needed.

For the camera WA, need apply according the silicon
stepping, no need WA since F1 stepping.

Change-Id: I97bf2337f92a36db4d852c7908c24dde6e08b1df
Tracked-On: PKT-2784
Signed-off-by: Yang, Dong <dong.yang@intel.com>
---
 drivers/gpu/drm/i915/i915_drv.h  | 4 ++--
 drivers/gpu/drm/i915/intel_fbc.c | 7 +++----
 2 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_drv.h b/drivers/gpu/drm/i915/i915_drv.h
index 668b9dd..197a831 100644
--- a/drivers/gpu/drm/i915/i915_drv.h
+++ b/drivers/gpu/drm/i915/i915_drv.h
@@ -2725,13 +2725,13 @@ static inline bool intel_vtd_active(void)
 
 static inline bool intel_scanout_needs_vtd_wa(struct drm_i915_private *dev_priv)
 {
-	return INTEL_GEN(dev_priv) >= 6 && intel_vtd_active();
+	return false;
 }
 
 static inline bool
 intel_ggtt_update_needs_vtd_wa(struct drm_i915_private *dev_priv)
 {
-	return IS_BXT_REVID(dev_priv, 0, BXT_REVID_D0) && intel_vtd_active();
+	return IS_BXT_REVID(dev_priv, 0, BXT_REVID_D0);
 }
 
 int intel_sanitize_enable_ppgtt(struct drm_i915_private *dev_priv,
diff --git a/drivers/gpu/drm/i915/intel_fbc.c b/drivers/gpu/drm/i915/intel_fbc.c
index 728a20e..2c09503 100644
--- a/drivers/gpu/drm/i915/intel_fbc.c
+++ b/drivers/gpu/drm/i915/intel_fbc.c
@@ -1280,10 +1280,9 @@ static int intel_sanitize_fbc_option(struct drm_i915_private *dev_priv)
 static bool need_fbc_vtd_wa(struct drm_i915_private *dev_priv)
 {
 	/* WaFbcTurnOffFbcWhenHyperVisorIsUsed:skl,bxt */
-	if (intel_vtd_active() &&
-	    (IS_SKYLAKE(dev_priv) || IS_BROXTON(dev_priv))) {
-		DRM_INFO("Disabling framebuffer compression (FBC) to prevent screen flicker with VT-d enabled\n");
-		return true;
+	if (intel_vtd_active() && IS_SKYLAKE(dev_priv)) {
+			DRM_INFO("Disabling framebuffer compression (FBC) to prevent screen flicker with VT-d enabled\n");
+			return true;
 	}
 
 	return false;
-- 
2.7.4

